### DISCUSSION TITLE
Long-term storage for recordings?

### DISCUSSION THREAD

**Discussion by HaLo2FrEeEk** [JurajNyiri/HomeAssistant-Tapo-Control#671] [General]
I'm running HA in a VM under Proxmox on a ThinkCentre mini PC. When I enabled the "enable media synchronization" for 6 cameras, I could hear the mini PC from the other side of the house. I don't know why but it was working *hard*. After 30 minutes I didn't see any files in the specified folder, which was on an SMB drive set to "media". It created the folders, so I know the path was right, but no files. Maybe I'm impatient, my cameras are set to loop record continuously to 32gb sd cards so maybe I just didn't want long enough?

Either way, I plan to buy an enclosure and stick a bunch of huge drives in there and connect it to the mini PC to make a DAS/NAS, I would like to use it for long-term local storage for the recordings. I don't want to have my mini pc running like a jet engine all the time though, any ideas why that would have happened?

Update: I think I might have figured...something out, at least. I had the wrong path, forgot the `/media` portion, but I think the plugin doesn't fail gracefully if it can't find the cold storage path, HA kept restarting after a period of fan activity. I fixed the path and set it to keep 24 hours. I'm not sure why my CPU goes up to 40% just downloading from the camera and uploading to the smb drive. I'm also not sure why the fan revs up so high for just 40%. I think what I'll do is set it to keep 6 hours in cold storage, but then I'll sync those files to another folder so that I have longer term storage. The 6 hour cold storage files will self-delete, but I'll keep the backups.

**Comment by JurajNyiri** [comment]
The downloading process is very complex. It doesn’t just copy the file since a proprietary process is needed. It essentially streams an encoded proprietary stream format and then needs to decode it into a standard once it records the intended chunk of data. 

As for the wrong path it should default to directory under config/.storage/tapo_control.

**Reply by HaLo2FrEeEk** [reply]
Ah, that explains the high CPU usage. Thank you for the clarification. I guess once it settles it shouldn't be an issue keeping the footage synchronized, it's just that I'm asking 6 cameras to sync 12 hours of footage.

I'm not sure why it wasn't working when I had the wrong path. I only had one camera with sync turned on, and the path was set to `/netstor1/Cameras/Cam1/`, I forgot the `/media` in front. Once I added that it worked.

I have, however, discovered something strange. I set the hours to keep to 6 and nothing came over. I set it to 12 and only the last few hours downloaded. My TZ is PST, so currently -7 hours, could this be affecting things?

I also have 1 camera that doesn't seem to want to download anything. I might, again, be being impatient, but all the other cameras have downloaded at least a few files so far.

I wish these cameras stored its files like Wyze cameras did. Each video was a 1 minute file, there were 60 files in an hour directory, inside of a date directory. Made drilling down to a specific time really easy. Would there be any way to accomplish something like that using this integration? I'm not knowledgeable enough about FFMpeg to make that work.

**Comment by HaLo2FrEeEk** [comment]
I've noticed a quirk, I think caused by the continuous recording on the camera. I'm getting duplicates in my cold storage. I wrote a quick little script to list the timestamp filenames as actual datetimes. Here's what I'm seeing, this is just one camera but I'm seeing it for all of them:

```
"8/31/2024, 09:16:40 - 09:17:53", (3.45 MB)
"8/31/2024, 09:16:40 - 09:28:31", (35 MB)
"8/31/2024, 09:16:40 - 09:44:45", (82.8 MB)
"8/31/2024, 09:16:40 - 10:31:31", (227 MB)
```

Looking back through the "Tapo Recordings" page in Media, I can see a file for 8:16:41 - 9:31:32, so I clearly also have a DST error in the timestamp filenames in cold storage. It seems like since it's continuously recording, it sees the currently-recording file as new every time it checks and streams and transcodes it from the beginning over and over again?

I've had all 6 cameras syncing 24 hours, for the last ~15 hours, and there are *still* gaps in the files from some of the cameras. The mini pc has been at 35-45% CPU the entire time.

This duplicate problem also messes up my plan to just copy the files from the cold storage path to a separate path for actual long term storage. In the example above, 120MB is wasted in the first 3 files. Not a lot, but if I was just copying everything it would build up fast.

Sorry if I'm being a bother, I'm very confused about all this but it's probably because I'm trying to use things in unintended ways. Thank you for the help!
